### ISSUE TITLE
argo-rollouts pod can't match virtualservice sourceLabels

### ISSUE THREAD

**Issue by labaq** [istio/istio#40034]
### Bug Description

my virtualservice spec is
```
spec:
  hosts:
    - example-demo02-stable
  http:
    - match:
        - sourceLabels:
            role: canary
      route:
        - destination:
            host: example-demo02-canary
            port:
              number: 8080
    - route:
        - destination:
            host: example-demo02-stable
            port:
              number: 8080
```
and my rollout have set labels to stable pod and canary pod like this
```
    canary:
      canaryMetadata:
        labels:
          role: canary
      stableMetadata:
        labels:
          role: stable
```
on first canary the pod have the wright labels so the rule is match
but when canary done, the canary pod will become stable pod and the label is change to stable, but it still match the canary rule, i think istio have already record this pod have canary label but not update to stable label.
can i have update the record for the right labels?

### Version

```prose
client version: 1.13.5
control plane version: 1.15
data plane version: 1.13.5 (8 proxies)

Client Version: v1.24.1
Server Version: v1.22.8+k3s1
```

**Comment by hzxuzhonghu** [comment]
I think i know what's the cause, you update the pod label at some time after it is stable, right? Unfortunately, istio does not update  proxy labels after startup.

**Comment by labaq** [comment]
sound so bad, i just found if i restart the pod it will match my virtualservice rule
so there is no way can i update the proxy labels? or any workaround can  do?

**Comment by hzxuzhonghu** [comment]
Take a look at my fix

**Comment by labaq** [comment]
i'm not familiar with istio code, how can i test it😅

**Comment by hzxuzhonghu** [comment]
Just build a new pilot discovery image and replace

**Comment by m-wynn** [comment]
We're still seeing this issue.  If I understand correctly, the fixes did not make it into the Istio 1.15 release.  Are there any updates we need to do to do to get this working in our environment, or can we expect to see it in an upcoming release?

**Comment by hzxuzhonghu** [comment]
right
